src/Compilers/CSharp/Portable/Binder/BindingDiagnosticBag.cs

```diff
@@ -95,19 +96,22 @@ internal void AddDependencies(Symbol? symbol)
 
         internal bool ReportUseSite(Symbol? symbol, SyntaxNode node)
         {
-            return ReportUseSite(symbol, node.Location);
+            return ReportUseSite(symbol, node, static node => node.Location);
         }
 
         internal bool ReportUseSite(Symbol? symbol, SyntaxToken token)
         {
-            return ReportUseSite(symbol, token.GetLocation());
+            return ReportUseSite(symbol, token, static token => token.GetLocation());
         }
 
         internal bool ReportUseSite(Symbol? symbol, Location location)
+            => ReportUseSite(symbol, location, static location => location);
```

Comment:
>=> ReportUseSite(symbol, location, static location => location); [](http://example.com/codeflow?start=12&length=64)

It might be worth avoiding allocating delegates on this code path (even cached ones).